core/src/main/java/org/elasticsearch/script/CompiledScript.java

```diff
@@ -24,17 +24,39 @@
  */
 public class CompiledScript {
 
+    private final ScriptService.ScriptType type;
+    private final String name;
     private final String lang;
     private final Object compiled;
 
     /**
      * Constructor for CompiledScript.
+     +     +     * @param type The type of script to be executed.
```

Comment:
Oops.  Merge error from using a patch between the pre and post plugin era.  Fixed.